mat64/index_bound_checks.go

```diff
@@ -35,3 +35,31 @@ func (m *Dense) set(r, c int, v float64) {
 	}
 	m.mat.Data[r*m.mat.Stride+c] = v
 }
+
+func (m *Vector) At(r, c int) float64 {
+	if c != 0 {
+		panic(ErrColIndexOutOfRange)
```

Comment:
If by differentiating, you mean the difference between row and column access, then we do differentiate in the Dense case just above. There isn't a named Error for it, but the plan is to add that as a named error. I had done so in Triangular, though looking now the error names are different, and it looks like the Triangular one is better (same message, more compact name).